pre-commit.sh

```diff
@@ -20,7 +20,7 @@
 # This file is git pre-commit hook.
 #
 # Soft link it as git hook under top dir of apache arrow git repository:
-# $ ln -s  ../../rust/pre-commit.sh .git/hooks/pre-commit
```

Comment:
Are these pre-commit changes intentional here? This seems unrelated to adding the `show` function.

Reply:
Yes, I found that pre-commit does not work, so I changed it, do I need to submit another PR?

Reply:
I would recommend a separate PR (mostly so we can tag the original author of the pre-commit and have a discussion about what to do there without holding up this PR)

Reply:
OK, I will open an issue and submit a PR to fix this.

Reply:
Just run `ln -s  ../../rust/pre-commit.sh .git/hooks/pre-commit` and exec `git commit`